### PR TITLE
Receive: Writer benchmark - address feedback from previous

### PR DESCRIPTION
**Pull request by matej-g**
Signed-off-by: Matej Gera <matejgera@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This PR adds some improvements and clarifications on the benchmark, in line with previous PR review, see https://github.com/thanos-io/thanos/pull/5533#pullrequestreview-1050573520.

## Verification
Benchmark results with default `benchtime`:
```
goos: linux
goarch: amd64
pkg: github.com/thanos-io/thanos/pkg/receive
cpu: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
BenchmarkWriterTimeSeriesWithSingleLabel_10-12           1602033              3733 ns/op             411 B/op          9 allocs/op
BenchmarkWriterTimeSeriesWithSingleLabel_100-12           298536             20916 ns/op             409 B/op          9 allocs/op
BenchmarkWriterTimeSeriesWithSingleLabel_1000-12           31250            190390 ns/op             420 B/op          9 allocs/op
BenchmarkWriterTimeSeriesWith10Labels_10-12              1269086              4589 ns/op             411 B/op          9 allocs/op
BenchmarkWriterTimeSeriesWith10Labels_100-12              191502             30629 ns/op             410 B/op          9 allocs/op
BenchmarkWriterTimeSeriesWith10Labels_1000-12              19999            288460 ns/op             429 B/op          9 allocs/op
PASS
ok      github.com/thanos-io/thanos/pkg/receive 77.922s
```

### CODE REVIEW

File: pkg/receive/writer_test.go [hunk 1]
```diff
@@ -237,12 +229,17 @@ func benchmarkWriter(b *testing.B, labelsNum int, seriesNum int) {
 	}
```
Comment (saswatamcode) [inline pkg/receive/writer_test.go:229]: Maybe we should check error here?

**Review by saswatamcode — APPROVED** [review]
Thanks! One tiny nit!

File: pkg/receive/writer_test.go (hunk 1 above)
Comment (bwplotka) [reply pkg/receive/writer_test.go:229]: Yes, it would be nice - no harm (little latency)

**Review by bwplotka — APPROVED** [review]
Thanks! 💪🏽 

Any reason for not checking err?

File: pkg/receive/writer_test.go (hunk 1 above)
Comment (matej-g) [reply pkg/receive/writer_test.go:229]: Sorry, I forgot to commit this change it looks, I'll fix it :+1:

File: pkg/receive/writer_test.go [hunk 2]
```diff
@@ -232,16 +225,22 @@ func benchmarkWriter(b *testing.B, labelsNum int, seriesNum int) {
 	b.ResetTimer()
 
 	for i := 0; i < b.N; i++ {
-		_ = w.Write(ctx, "foo", wreq)
+		err := w.Write(ctx, "foo", wreq)
```
Comment (bwplotka) [inline pkg/receive/writer_test.go:228]: could be one line 🙈

**Review by bwplotka — APPROVED** [review]
Thanks!

File: pkg/receive/writer_test.go (hunk 2 above)
Comment (matej-g) [reply pkg/receive/writer_test.go:228]: Fixed :wink:

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**